assets/js/src/integrations/wp-consent-api.js

```diff
@@ -0,0 +1,56 @@
+const consentMap = {
+	statistics: [ 'analytics_storage' ],
+	marketing: [ 'ad_storage', 'ad_user_data', 'ad_personalization' ],
+};
+
+export const setCurrentConsentState = ( {
+	tracker_function_name: trackerFunctionName,
+} ) => {
+	// eslint-disable-next-line camelcase -- `wp_has_consent` is defined by the WP Consent API plugin.
+	if ( typeof wp_has_consent === 'function' ) {
+		if ( window.wp_consent_type === undefined ) {
+			window.wp_consent_type = 'optin';
+		}
+
+		const consentState = {};
+
+		for ( const [ category, types ] of Object.entries( consentMap ) ) {
+			// eslint-disable-next-line camelcase, no-undef
+			if ( wp_has_consent( category ) ) {
+				types.forEach( ( type ) => {
+					consentState[ type ] = 'granted';
+				} );
+			}
```

Comment:
:nail_care: We can add an explanation to this `eslint-disable-next-line` as well.

:question: Related to https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/425#discussion_r1608711163

```suggestion
			types.forEach( ( type ) => {
				// eslint-disable-next-line camelcase, no-undef
				consentState[ type ] = wp_has_consent( category )
					? 'granted'
					: 'denied';
			} );
```

AFAIUnderstood the docs `gtag('consent', 'update', ... )` is for the current visitor, so the region does not matter. If they disallow, I assume they mean to disallow tracking their data regardless where they connect from at the moment.

Reply:
Apologies if I wasn't clear in my earlier description @tomalec.

If GA4W is installed on a store (_without the default consent state being filtered_) then for visitors from the EEA region, the on-page default will be `denied`.

A user from a different country then visits the store and the on-page default does not apply to them.

With this suggested change, for visitors from outside the EEA region (_that the on-page default doesn't apply to_) we would still send an update denying all categories even if they haven't interacted with a consent banner because `wp_has_consent` will default to `false`.

Here are two screenshots taken while visiting the site from the USA via a VPN with the suggested change without interacting with a consent banner:

**On-page default:**

<img width="1427" alt="Screenshot 2024-05-28 at 11 39 10" src="https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/40762232/8bcdb125-a578-412b-a15b-fecd98eb46cd">

**On-page update:**

<img width="1461" alt="Screenshot 2024-05-28 at 11 39 17" src="https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/40762232/9781228b-08f3-46e6-88c4-e4d6a8bd117d">

So here no traffic from outside of the EEA region will be tracked and merchants might not display a banner to those users to change the consent state if it's not a local requirement.

I've adjusted the code to send the entire consent state but only if the category cookie is set to avoid sending the update before a user has interacted with a consent banner. 86e43c7

Reply:
Thanks for the explanations, now I get it and agree :)